pkg/kvstore/etcd_tls.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kvstore
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+
+	k8sclient "github.com/cilium/cilium/pkg/k8s/client"
+
+	"github.com/coreos/etcd-operator/pkg/util/etcdutil"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type tlsData struct {
+	certData []byte
+	keyData  []byte
+	caData   []byte
+}
+
+func getTLSDataFromSecret(kubecli kubernetes.Interface, ns, se string) (*tlsData, error) {
+	secret, err := kubecli.CoreV1().Secrets(ns).Get(se, v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return &tlsData{
+		certData: secret.Data[etcdutil.CliCertFile],
```

Comment:
Let's define constants for this as this is a stable API. No need to add dependencies.